src/test/scala/nl.knaw.dans.easy.multideposit/BlackBoxSpec.scala

```diff
@@ -66,80 +92,159 @@ class BlackBoxSpec extends UnitSpec with BeforeAndAfter with MockFactory with Cu
+            .filterNot(_ contains "bag-info.txt")
+            .filterNot(_ contains "manifest-sha1.txt").toSeq
+      }
+
+      it should "check the files in data/" in {
+        doNotRunOnTravis()
+
+        val dataDir = bag.resolve("data/")
+        dataDir.toFile should exist
+        dataDir.listRecursively().map {
+          case file if Files.isDirectory(file) => file.getFileName.toString + "/"
+          case file => file.getFileName.toString
+        } should contain theSameElementsAs dataContent
+      }
+
+      it should "check the files in metadata/" in {
+        doNotRunOnTravis()
+
+        managed(Files.list(bag.resolve("metadata")))
+          .acquireAndGet(_.iterator().asScala.toList)
+          .map(_.getFileName.toString) should contain only("dataset.xml", "files.xml")
+      }
+
+      it should "check metadata/dataset.xml" in {
+        doNotRunOnTravis()
+
+        def removeElemByName(label: String) = new RuleTransformer(new RewriteRule {
+          override def transform(n: Node): Seq[Node] = {
+            n match {
+              case e: Elem if e.label == label => NodeSeq.Empty
+              case e => e
+            }
+          }
+        })
+
+        val datasetXml = bag.resolve("metadata/dataset.xml")
+        val expDatasetXml = expBag.resolve("metadata/dataset.xml")
+        val datasetTransformer = removeElemByName("available")
+
+        // skipping the available field here
+        datasetTransformer.transform(XML.loadFile(datasetXml.toFile)) should
+          equalTrimmed(datasetTransformer.transform(XML.loadFile(expDatasetXml.toFile)))
+      }
+
+      it should "check metadata/files.xml" in {
+        doNotRunOnTravis()
+
+        val filesXml = bag.resolve("metadata/files.xml")
+        val expFilesXml = expBag.resolve("metadata/files.xml")
+
+        XML.loadFile(filesXml.toFile).toSet should equalTrimmed(XML.loadFile(expFilesXml.toFile).toSet)
+      }
+
+      it should "check deposit.properties" in {
+        doNotRunOnTravis()
+
+        val props = settings.outputDepositDir.resolve(s"allfields-$bagName/deposit.properties")
+        val expProps = expectedOutputDir.resolve(s"input-$bagName/deposit.properties")
+
+        // skipping comment lines, as well as the line with randomized bag-id
+        props.read().lines.toSeq should contain allElementsOf
+          expProps.read().lines
+            .filterNot(_ startsWith "#")
+            .filterNot(_ contains "bag-store.bag-id")
+            .toSeq
+      }
     }
+
+    "ruimtereis01" should behave like bagContents("ruimtereis01", expectedDataContentRuimtereis01)
+    "ruimtereis02" should behave like bagContents("ruimtereis02", expectedDataContentRuimtereis02)
+    "ruimtereis03" should behave like bagContents("ruimtereis03", expectedDataContentRuimtereis03)
+    "ruimtereis04" should behave like bagContents("ruimtereis04", expectedDataContentRuimtereis04)
   }
 
+  beforeAll()
+
+  "allfields" should behave like allfieldsSpec()
```

Comment:
Not sure what you mean, but maybe that's the bit your confused about 😉  Let me explain: there are two tests I want to perform: one for the `allfields` test resource, which is a correctly filled in `instructions.csv` with a correct file set, and one for the `invalidCSV` test resource, which (as the name suggests) has errors in its `instructions.csv`.

In the `allfields` test I want to make sure that every file from all deposits inside the resulting multi-deposit is equal/equivalent to the expected output. Therefore, for each deposit I need to do a number of tests. What I could have done is just write them all out, while copy-pasting all 10 tests for each of the deposits (currently there are 4, so 40 tests, all duplicated, but I foresee more of them coming in the future) and fix a test in multiple places when I made a mistake. Of course we agree that this is not desirable!

To fix this I put them in a function that can be called using the `should behave like` construct in ScalaTest. (Note that `FlatSpec` doesn't allow for nested tests, so that's why I can't put the outside function inside the test on line 246. This function once performs the creation of the deposits by running `Main.run` and then runs all 40 tests. To really get duplication away and not run any setup code (such as `Main.run`) twice, I had to wrap the 'real' tests in another function and give that function some arguments to abstract over the variables (which is the deposit's name and the list of expected files/folders) and call this function four times with different arguments. This keeps the 'real' tests simple and descriptive in the `it should "..." in {...}`, makes them easy to read and makes them be able to fail on there own without influencing the other tests.

To recapitulate: the spec runs the `allfields` test, which calls the `allfieldsSpec` function, which calls `bagContents` one time for each of the resulting deposits.

Since the `invalidCSV` tests isn't as complicated and doesn't require nested tests, I just used the regular way of testing.